R/compact_character.R

```diff
@@ -0,0 +1,20 @@
+#' Remove empty strings from character
+#'
+#' @param x A single character or a vector of characters.
+#'
+#' @return
+#'
+#' A character or a character vector with empty strings removed.
+#'
+#' @examples
+#' compact_character(c("x", "y", NA))
+#' compact_character(c("x", NULL, "", "y"))
+#' compact_character(list("x", "y", NA))
+#' compact_character(list("x", NULL, "", "y"))
+#'
+#' @export
+
+compact_character <- function(x) {
+  x[!sapply(x, function(i) nchar(i) == 0 || is.null(i) || any(i == "NULL", na.rm = TRUE))]
```

Comment:
```suggestion
  x[!sapply(x, function(i) nchar(i) == 0 || is.null(i) || all(is.na(i)) || any(i == "NULL", na.rm = TRUE))]
```

Reply:
Or maybe any() there?

Reply:
> I would expect NAs to be removed too, no?

Yeah, me too, but @strengejacke has been using this internal helpers for a long time, so I am not sure if this is deliberate behavior or something that has gone unnoticed thus far.

Reply:
I think we can have `any()` here. Not sure if it will break code, but I think we can just "re-write" this from scratch.